Remove Apex feature flag & delete CLI code path (#3148)
forcedotcom/salesforcedx-vscode: docs/_articles/ja/user-guide/perf-enhancements.md

lang: en
---

> The Apex, Deploy and Retrieve commands are in the process of moving to a new architecture. These commands have been rearchitected into independent Typescript libraries and CLI plugins as part of our larger effort to break up the salesforce-alm plugin. [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html). There’s many great benefits to this effort and one of the best is better performance. For single file deploys and retrieves, you'll likely notice a significant performance gain when using the latest architecture.

Reviewer: ```suggestion
The Deploy and Retrieve commands are in the process of moving to a new architecture. We moved these commands to independent Typescript libraries and CLI plugins as part of our larger effort to break up the salesforce-alm plugin. [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html). There’s many great benefits to this effort and one of the best is better performance. For single file deploys and retrieves, you'll likely notice a significant performance gain when using the latest architecture.
```

> The Apex, Deploy and Retrieve commands are in the process of moving to a new architecture. These commands have been rearchitected into independent Typescript libraries and CLI plugins as part of our larger effort to break up the salesforce-alm plugin. [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html). There’s many great benefits to this effort and one of the best is better performance. For single file deploys and retrieves, you'll likely notice a significant performance gain when using the latest architecture.

Author: ```suggestion
The Apex and Deploy/Retrieve commands are in the process of moving to a new architecture. We moved these commands to independent Typescript libraries and CLI plugins as part of our larger effort to break up the salesforce-alm plugin. [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html). There’s many great benefits to this effort and one of the best is better performance. For single file deploys and retrieves, you'll likely notice a significant performance gain when using the latest architecture.
```

4. CLI updated to use the library. The VS Code setting is then retired. Both the CLI and VS Code extensions are using the library.

|                         | Library complete | Available to try in VS Code | On by default in VS Code | CLI Updated |
| ----------------------- | :--------------: | :-------------------------: | :----------------------: | :---------: |
| Apex Library            |        ✔️        |             ✔️              |            ✔️            |     ✔️      |
| Deploy Retrieve Library |        ✔️        |             ✔️              |            ✔️            |             |

## Setup

The Apex library has been fully implemented, but the Deploy/Retrieve library has a corresponding VS Code setting. Access the Deploy/Retrieve setting by selecting Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS). Setting to false means that VS Code uses the CLI to execute those commands.

Within settings, you will see the following:

1. _Experimental: Deploy Retrieve_

Known gaps:

> - If you're using [CLI Plug-In Hooks](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks.htm), those are not available when using the library from VS Code. Uncheck the settings above to use the CLI directly for those commands.

Reviewer: ```suggestion
- If you're using [CLI Plug-In Hooks](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks.htm), they are not available when using the library from VS Code. Uncheck the settings above to use the CLI directly for those commands.
```
